packages/auth-remix/src/server.ts

```diff
@@ -0,0 +1,795 @@
+            }
+
+            const headers = new Headers();
+            headers.append(
+              "Set-Cookie",
+              `${this.options.authCookieName}=${tokenData.auth_token}; HttpOnly; SameSite=strict; Path=/`
+            );
+
+            return oathCallbackCall(onEmailVerify, headers, {
+              error: null,
+              tokenData,
+            });
+          }
+
+          case "signout": {
+            if (!onSignout) {
+              throw new Error(`'onSignout' auth route handler not configured`);
+            }
+
+            const headers = new Headers({
+              "Set-Cookie": `${
+                this.options.authCookieName
+              }=; HttpOnly; SameSite=strict; Expires=${Date.now()}; Path=/`,
+            });
+
+            return oathCallbackCall(onSignout, headers);
+          }
+
+          default:
+            throw new Error("Unknown auth route");
+        }
+      },
+    };
+  }
+
+  async emailPasswordSignUp(
+    req: Request,
+    data?: { email: string; password: string }
+  ): Promise<{ tokenData: TokenData; headers: Headers }>;
+  async emailPasswordSignUp<Res extends Response>(
+    req: Request,
+    cb: (params: ParamsOrError<{ tokenData: TokenData }>) => Res | Promise<Res>
+  ): Promise<Res>;
+  async emailPasswordSignUp<Res extends Response>(
+    req: Request,
+    data: { email: string; password: string },
+    cb: (params: ParamsOrError<{ tokenData: TokenData }>) => Res | Promise<Res>
+  ): Promise<Res>;
+  async emailPasswordSignUp(
+    req: Request,
+    dataOrCb?:
+      | { email: string; password: string }
+      | ((
+          params: ParamsOrError<{ tokenData: TokenData }>
+        ) => Response | Promise<Response>),
+    cb?: (
+      params: ParamsOrError<{ tokenData: TokenData }>
+    ) => Response | Promise<Response>
+  ): Promise<
+    | Response
+    | {
+        tokenData: TokenData;
+        headers: Headers;
+      }
+  > {
+    return handleAction(
+      async (data, headers) => {
+        const [email, password] = _extractParams(
+          data,
+          ["email", "password"],
+          "email or password missing"
+        );
+
+        const result = await (
+          await this.core
+        ).signupWithEmailPassword(
+          email,
+          password,
+          `${this._authRoute}/emailpassword/verify`
+        );
+
+        headers.append(
+          "Set-Cookie",
+          `${this.options.pkceVerifierCookieName}=${result.verifier}; HttpOnly; SameSite=strict`
+        );
+
+        if (result.status === "complete") {
+          const tokenData = result.tokenData;
+
+          headers.append(
+            "Set-Cookie",
+            `${this.options.authCookieName}=${tokenData.auth_token}; HttpOnly; SameSite=strict`
+          );
+          return { tokenData, headers };
+        }
+
+        return { tokenData: null, headers };
+      },
+      req,
+      dataOrCb,
+      cb
+    );
+  }
+
+  async emailPasswordResendVerificationEmail(
```

Comment:
This action doesn't technically need the request object, but maybe it should still have it for consistency?

Reply:
Also when we fix this: https://github.com/edgedb/edgedb/issues/6503, then the action will need to set the pkce verifier cookie, so we should probably just make this action match the others (with the callback overloads, etc) now anyway.